### ISSUE TITLE
could you make it any harder please?

### ISSUE THREAD

**Issue by kimballjohnson** [chandermani/angular6byexample#11]
You know, it's very hard to jump back and forth between a book page on screen and the code and match up all the ways different authors refer to and explain away about tech issues using their own manner of speech and way of thinking.

But when they take the source and fragment it and stuff it into a hundred or more folders buried six and seven layers deep, it's really beyond understanding!

If you have to maintain every step as a separate branch rather than just building a single version of your example, layer by layer that matches the flow of the instructions in your book, then please keep your source in top-level folders only, each one containing the source for each branch (checkpoint? - you have to invent nomenclature?). 
It took me over an hour to unwind it correctly!!

![UnscrewedVersioning](https://user-images.githubusercontent.com/2092723/62333585-f6ba7200-b478-11e9-9fae-143c4577d83a.png)

**Comment by chandermani** [comment]
I agree with your problem @kimballjohnson. Single companion download does not work. 
I would instead suggest you use the git repo with branches which does not come with these complexities. 
The github branches (checkpoints) help the reader to compare and understand how the application has evolved, with the added bonus of starting afresh from a new checkpoint if one gets stuck.
